### ISSUE TITLE
Export preset to selected file

### ISSUE THREAD

**Issue by byron-hawkins** [wwmm/easyeffects#2523]
Creating a preset currently puts it in a fixed location, for example `~/.config/easyeffects`. For users wanting to regularly create and update presets for various scenarios, the workflow is confusing and awkward. Suppose a preset is loaded, then settings are changed. Is the preset automatically changed to reflect the new active settings? If not, how can it be updated with the changes? Or, how can a new version of the preset be created? 

Instead, it would be nicer to have an optional mode where a preset can only be saved from the current settings into a selected file. In this mode, there is no such thing as a named preset, and the `~/.config/easyeffects` directory is never used. Instead, there is a `save preset` button allowing the user to select a file, which is written (or over-written) with a preset reflecting the current settings.

If this is possible in the current version, please let me know what I missed! Otherwise, I hope you can consider it as a feature request.

**Comment by wwmm** [comment]
> Suppose a preset is loaded, then settings are changed. Is the preset automatically changed to reflect the new active settings? If not, how can it be updated with the changes? Or, how can a new version of the preset be created?

No. Settings are saved to our `GSettings` database. That is how they are reloaded when EasyEffects is restarted. It has no relation to the preset files. They are just a quick way to load a given set of values. You have to manually save (overwrite) them to update their contents.

>it would be nicer to have an optional mode where a preset can only be saved from the current settings into a selected file. In this mode, there is no such thing as a named preset

But this is what happens. When you click on the save button the current settings are saved to the preset file. As I explained above there is no need to create preset files to restore the last used settings.

**Comment by byron-hawkins** [comment]
There is no save button in my version of Easy Effects. All I can ever do is create a preset by name, and thereafter, it shows the preset as somehow "active". I would like to have the save button you mentioned. Does this exist in some other version? I have 7.0.6 on Fedora 38.

**Comment by wwmm** [comment]
>There is no save button in my version of Easy Effects. 

There is. It is the button between the `load` and the `trash icon` of the button that deletes presets. When you click on it the current preset settings are overwritten.

**Comment by byron-hawkins** [comment]
Well ok, but it saves to a pre-defined folder. I was hoping for a save button (or a mode in which there is a save button) that opens a "save as..." dialog so I can put the file exactly where it goes in my workflow. Creating a preset to get a preset file is awkward, and pulling a preset file out of `~/.config/` is awkward. I'd like it to work **like a word processor**, where a preset is like a text file. 

I realize this is probably not what everyone wants. Just a suggestion.

**Comment by servimo** [comment]
You can load a preset, then give it a name, press the "+". It's the same preset you loaded with the name you gave.

**Comment by wwmm** [comment]
>I realize this is probably not what everyone wants. Just a suggestion.

No problems. Dealing with a fixed folder location makes the code a lot simpler. than having to searching for different locations the user may be desiring to use. This also would probably cause some problems for the Flatpak package as there are restriction to where the application is allowed to have write access.

**Comment by wwmm** [comment]
I may have misunderstood what you want. Do you just want an easy way to export all the presets to a given location or do you expect EasyEffects to use the presets in an arbitrary user location? Just copying the files to a destination path may be doable. But making EasyEffects using what is in a custom path lie it does to what is inside`~/.config/easyeffects/` is not practical.

**Comment by byron-hawkins** [comment]
Just a plain export would simplify things 90% for me. It would make Easy Effects similar to the other audio tools I'm familiar with, where usually there is some internal "staging area" that stores files representing the active state. If the user wants to import/export manually, that can be done with a separate button and/or menu item. It's not exactly what I would choose, but it works well enough.

**Comment by cottton** [comment]
Just tried to export my presets ... searched the "export" button ... because where you can import you should be able to export ...
no export found :/
Sadface.

**Comment by wwmm** [comment]
> Just tried to export my presets ... searched the "export" button ... because where you can import you should be able to export ... no export found :/ Sadface.

I will try to remember to solve this in the Qt port.

**Comment by cottton** [comment]
@wwmm I just wanna mention: i found my presets in the files, but in a totally unexpected path: `/home/{username}/.var/app/com.github.wwmm.easyeffects/config/easyeffects/output/`

**Comment by wwmm** [comment]
> @wwmm I just wanna mention: i found my presets in the files, but in a totally unexpected path: `/home/{username}/.var/app/com.github.wwmm.easyeffects/config/easyeffects/output/`

This is normal when installing through a Flatpak package. Its paths are different from the usual. That is why I think we have to implement an export function. Flatpak paths are not obvious for many people.

**Comment by wwmm** [comment]
Implemented in our master branch. The Presets menu now has a button that copies all presets of the corresponding pipeline to a folder the user selects through a directory dialog. I do not know yet if the current approach is working on Flatpak. Feedback on that is welcome.
